src/components/boolean-search/boolean-table.tsx

```diff
@@ -0,0 +1,307 @@
+import React, { useEffect, useState, useMemo } from 'react';
+import PropTypes from 'prop-types';
+import { useTable } from 'react-table';
+import { windowRef } from '@isrd-isi-edu/deriva-webapps/src/utils/window-ref';
+
+const BooleanTable = ({
+  rows,
+  setRows,
+  strength,
+  pattern,
+  location,
+  stageFrom,
+  source,
+  changeFiltersDisplayText,
+  addClicked,
+  setAddClicked
+}: any) : JSX.Element=> {
```

Comment:
We should define our own type for this component instead of using `any`. This example is how we have done it in most of our components in chaise too:
https://github.com/informatics-isi-edu/deriva-webapps/blob/master/src/components/search-input.tsx#L10-L38